### DISCUSSION TITLE
Data not showing on other components

### DISCUSSION THREAD

**Discussion by Kunal5200** [vercel/next.js#45841] [Help]
When I pass the API response to the router. push query string it shows empty on another component.
1st component
  router.push({
          pathname: '/dashboard',
          query: { userdetails: user },
        })

2nd component
const Userdashboard = () => {
  const router = useRouter()
  const {userdetails}=router.query
  console.log(userdetails)
  return <div>userdetails</div>
}

console of the browser
![image](https://user-images.githubusercontent.com/112200812/218392795-ebb08b16-90a4-4cce-8a82-0d5a7f9e754f.png)

**Comment by icyJoseph** [comment]
Hi,

Are you trying to pass anything other than strings?

What shape does `user`, API response as you call it?

You cannot pass very complex data structures through queries, at best strings or list of strings.
